src/daft-algebra/src/boolean.rs

```diff
@@ -22,3 +24,161 @@ pub fn split_conjunction(expr: &ExprRef) -> Vec<ExprRef> {
 pub fn combine_conjunction<T: IntoIterator<Item = ExprRef>>(exprs: T) -> Option<ExprRef> {
     exprs.into_iter().reduce(|acc, e| acc.and(e))
 }
+
+/// Converts a boolean expression to conjunctive normal form (AND of ORs)
+pub fn to_cnf(expr: ExprRef) -> ExprRef {
+    let dnf_form = to_dnf(expr.not()).not();
```

Comment:
thoughts on doing to_cnf directly instead of modified version of to_dnf? I don't know how many expressions there are / performance concerns. but looks good so will trust you on this

Reply:
The standard algorithm for converting to CNF is to first convert its negation to DNF, so I figured I would also just expose a to_dnf as well in case we ever need it.

Reply:
ok sounds good to me